cmd/kube-proxy/app/server.go

```diff
@@ -237,6 +237,9 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
 		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
 		iptables.CleanupLeftovers(iptInterface)
 	}
+	if err != nil && proxier == nil {
```

Comment:
should this be `err != nil || proxier == nil` since line 243 will fail if proxier is nil? (even if err is nil, proxier should not be nil)